runtime/include/cloe/databroker.hpp

```diff
@@ -0,0 +1,1254 @@
+        return get_value_fn;
+    }
+#if defined (__HAVE_BETTER_ERROR_MESSAGES) && (__HAVE_BETTER_ERROR_MESSAGES > 0)
+    /**
+     * Compiler trap to provide a meaningful error-message
+     */
+    template < typename TValue >
+    typename enable_if_is_unwanted<TValue, const typed_get_value_function_t<TValue>*>::type getter() const
+    {
+        cause_compiler_error_due_to_unwanted_type<TValue>();
+    }
+#endif // defined (__HAVE_BETTER_ERROR_MESSAGES) && (__HAVE_BETTER_ERROR_MESSAGES > 0)
+
+    /**
+     * @brief Returns the current value of the signal
+     * @tparam TValue Type of the signal
+     * @return const TValue* Current value of the signal
+     */
+    template<typename TValue>
+    typename enable_if_is_wanted<TValue, const TValue*>::type value() const
+    {
+        validate_type<TValue>();
+        const typed_get_value_function_t<TValue>* getter_fn = getter<TValue>();
+        if (!getter_fn)
+        {
+            throw new std::logic_error("INTERNAL ERROR");
+        }
+        const TValue* value = getter_fn->operator()();
+        return value;
+    }
+#if defined (__HAVE_BETTER_ERROR_MESSAGES) && (__HAVE_BETTER_ERROR_MESSAGES > 0)
+    /**
+     * Compiler trap to provide a meaningful error-message
+     */
+    template < typename TValue >
+    typename enable_if_is_unwanted<TValue, const TValue*>::type value()
+    {
+        cause_compiler_error_due_to_unwanted_type<TValue>();
+    }
+#endif // defined (__HAVE_BETTER_ERROR_MESSAGES) && (__HAVE_BETTER_ERROR_MESSAGES > 0)
+
+    /**
+     * @brief Returns the getter function of the signal
+     * @tparam TValue Type of the signal
+     * @return Getter function of the signal
+     */
+    template<typename TValue>
+    typename enable_if_is_wanted<TValue, const typed_set_value_function_t<TValue>*>::type setter() const
+    {
+        validate_type<TValue>();
+        const typed_set_value_function_t<TValue>* set_value_fn = std::any_cast<typed_set_value_function_t<TValue>>(&set_value_);
+        return set_value_fn;
+    }
+#if defined (__HAVE_BETTER_ERROR_MESSAGES) && (__HAVE_BETTER_ERROR_MESSAGES > 0)    
+    /**
+     * Compiler trap to provide a meaningful error-message
+     */
+    template < typename TValue >
+    typename enable_if_is_unwanted<TValue, const typed_set_value_function_t<TValue>*>::type setter() const
+    {
+        cause_compiler_error_due_to_unwanted_type<TValue>();
+    }
+#endif // defined (__HAVE_BETTER_ERROR_MESSAGES) && (__HAVE_BETTER_ERROR_MESSAGES > 0)
+
+    /**
+     * @brief Sets the value of the signal
+     * @tparam TValue Type of the signal
+     * @param value Value of the signal
+     */
+    template <typename TValue>
+    typename enable_if_is_wanted<TValue, void >::type set_value(const TValue* value) const
+    {
+        validate_type<TValue>();
+        const typed_set_value_function_t<TValue>* setter_fn = setter<TValue>();
+        if (!setter_fn)
+        {
+            throw new std::logic_error("INTERNAL ERROR");

```

Comment:
Something like `should never happen: expect setter_fn to be set` is better than "INTERNAL ERROR", would you not agree? Or use assert to make clear that we are responsible for it.